brigade/core/inventory.py

```diff
@@ -168,6 +171,46 @@ def nos(self):
         """Network OS the device is running. Defaults to ``brigade_nos``."""
         return self.get("brigade_nos")
 
+    def ssh_connection(self, ssh_config_file=None, ssh_key_file=None):
+        if hasattr(self, "_ssh_connection"):
+            return self._ssh_connection
+
+        if ssh_config_file is None:
+            ssh_config_file = os.path.join(os.path.expanduser("~"), ".ssh", "config")
+
+        client = paramiko.SSHClient()
+        client._policy = paramiko.WarningPolicy()
+        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        ssh_config = paramiko.SSHConfig()
+        if os.path.exists(ssh_config_file):
+            with open(ssh_config_file) as f:
```

Comment:
Perhaps this should be handled in the same way as self._ssh_connection. Just to avoid reading the same file multiple times for multiple hosts, or have it auto loaded with the config.

Reply:
true

Reply:
actually, it is not that easy to solve here right now but it should be fixable once we have the `config` object you are working on

Reply:
I was just thinking about some function which would only load the file if it hadn't been previously loaded. But yes the config object would be a good start for this. Should need quite a few hosts before this actually is a problem, it's more that it's unnecessary.

Reply:
yeah, I think we should pass around the `Config` to all the hosts and solve it there so no need to bother for now. It's a minor optimization anyway.